unity-ggjj/Assets/Scripts/TextDecoder/Preloading/ObjectPreloader.cs

```diff
@@ -99,7 +99,7 @@ private void LoadEvidence(string evidenceName)
         }
         catch (NullReferenceException exception)
         {
-            Debug.LogWarning($"{exception.GetType().Name}: Object at path {path} could not be loaded");
+            throw new NullReferenceException($"{exception.GetType().Name}: Object at path {path} could not be loaded");
```

Comment:
Nit: Not passing **the exception we're capturing** into **the new exception we're throwing**, means we discard the stack of the original exception.
  
The "deepest" line we'll see in the console with the current version of the code is  
`unity-ggjj/Assets/Scripts/TextDecoder/Preloading/ObjectPreloader.cs:102`
However with...
```suggestion
            throw new NullReferenceException($"{exception.GetType().Name}: Object at path {path} could not be loaded", exception);
```
...we'll see that this exception is thrown because of something in the `try` block. For example:  
`unity-ggjj/Assets/Scripts/TextDecoder/Preloading/ObjectPreloader.cs:94` or maybe even  
`unity-ggjj/Assets/Scripts/TextDecoder/Preloading/ObjectStorage.cs:58` if the `.Contains()` would be able to throw.

I just realized this a few days ago, so that's why I've haven't been pointing this out in previous PRs. 👀

Reply:
Weirdly this is something I do habitually in Java, but not in C# apparently. Good catch!